utils.js

```diff
@@ -4255,6 +4257,217 @@ export async function parseSetupPyFile(setupPyData) {
   return await parseReqFile(lines.join("\n"), false);
 }
 
+function createPurlTemplate(packageData) {
+  const purlTemplate = `pkg:${packageData["kind"]}/${packageData["name"]}@${packageData["version"]}-${packageData["build"]}?os=${packageData["subdir"]}`;
```

Comment:
Any ideas if we might get undefined for any of these attributes or would pixi always return a value for all of these?

Reply:
I am certain it should be always present [lockfile satisfiability](https://pixi.sh/latest/features/lockfile/#lockfile-satisfiability). However, I asked the same in pixi project discussion, if it is possible for any of these values to be undefined I will change the `createPurlTemplate` function.